ocean_provider/test/test_config.py

```diff
@@ -50,3 +50,66 @@ def test_config_dict():
     }
     config = Config(options_dict=config_dict)
     assert config.aquarius_url == "https://another-aqua2.url"
+
+
+@pytest.mark.unit
+def test_allow_non_public_ip():
+    config_dict = {
+        "eth-network": {},
+        "resources": {
+            "aquarius.url": "https://another-aqua2.url",
+            "allow_non_public_ip": "False",
+        },
+    }
+    config = Config(options_dict=config_dict)
+    assert config.allow_non_public_ip == False
+
+    config_dict = {
+        "eth-network": {},
+        "resources": {
+            "aquarius.url": "https://another-aqua2.url",
+            "allow_non_public_ip": "0",
+        },
+    }
+    config = Config(options_dict=config_dict)
+    assert config.allow_non_public_ip == False
+
+    config_dict = {
+        "eth-network": {},
+        "resources": {
+            "aquarius.url": "https://another-aqua2.url",
+            "allow_non_public_ip": 0,
+        },
+    }
+    config = Config(options_dict=config_dict)
+    assert config.allow_non_public_ip == False
+
+    config_dict = {
+        "eth-network": {},
+        "resources": {
+            "aquarius.url": "https://another-aqua2.url",
+            "allow_non_public_ip": True,
+        },
+    }
+    config = Config(options_dict=config_dict)
+    assert config.allow_non_public_ip == True
+
+    config_dict = {
+        "eth-network": {},
+        "resources": {
+            "aquarius.url": "https://another-aqua2.url",
+            "allow_non_public_ip": "True",
+        },
+    }
+    config = Config(options_dict=config_dict)
+    assert config.allow_non_public_ip == True
+
+    config_dict = {
+        "eth-network": {},
+        "resources": {
+            "aquarius.url": "https://another-aqua2.url",
+            "allow_non_public_ip": 1,
+        },
+    }
```

Comment:
Please add a test case where `allow_non_public_ip` is not set (and thus would fallback to `None`).  

From a design standpoint - what should happen if `allow_non_public_ip` is not set?  I'm thinking when `allow_non_public_ip` is not set, then non-public IPs should not be allowed.

Reply:
this should be enough:  https://github.com/oceanprotocol/provider/blob/v4main/Dockerfile#L52